File: study/src/study/day24.java
package study;

public class day24 {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        String[] map1 = toBinary(n, arr1);
        String[] map2 = toBinary(n, arr2);

        for(int i=0; i<n; i++){
            String[] arrTmp1 = map1[i].split("");
            String[] arrTmp2 = map2[i].split("");

            String strTmp = "";
            for(int j=0; j<n; j++){
                if (arrTmp1[j].equals("0") && arrTmp2[j].equals("0")) {
                    strTmp = strTmp + " ";
                } else {
                    strTmp = strTmp + "#";
                }
            }
            answer[i] = strTmp;
        }

        return answer;
    }

    public String[] toBinary (int n, int[] map){
        String[] result = new String[n];
        for(int i=0; i<n; i++){
            result[i] = String.format("%"+n+"s", Integer.toBinaryString(map[i])).replace(" ", "0");
        }

        return result;
    }

    public static void main(String[] args){
        day24 a = new day24();
        int[] arr1 = {9, 20, 28, 18, 11};
        System.out.println(a.toBinary(5, arr1));
    }
}

File: study/bin/study/day24.py
def solution(n, arr1, arr2):
    answer = []
    map1 = toBinary(n, arr1)
    map2 = toBinary(n, arr2)

    for i in range(n):
        tmp1 = list(map1[i])
        tmp2 = list(map2[i])

        strTmp = ""
        for j in range(n):
            if tmp1[j]=='0' and tmp2[j]=='0':
                strTmp = strTmp + ' '
            else:
                strTmp = strTmp + '#'
        answer.append(strTmp)
    return answer

def toBinary(n, map):
    result = []
    for i in range(n):
        formatStr = "{:0" + str(n)+"b}"
        tmp = formatStr.format(map[i])
        result.append(tmp)
    return result

toBinary(5, [9, 20, 28, 18, 11])
solution(5,[9, 20, 28, 18, 11],[30, 1, 21, 17, 28])